routes/expenseRoute.js

```diff
@@ -56,7 +56,7 @@ router.patch('/:id', getExpense, async (req, res) => {
 //Deleting One
 router.delete('/:id', getExpense, async (req, res) => {
     try {
-        await res.expense.remove()
+        await res.expense.findByIdAndRemove(req.params.id)
```

Comment:
**note:** We need to be sure we are using the correct method from Mongoose.  `.remove()` drops the entire table 😱

Reply:
ahhh okay that makes sense. I'm only just realizing how many different commands mongoose comes with

Reply:
So I'm thinking for patch right, would I just do it in a similar way as the delete button with find by ID but just call the patch function instead? But also I'd use some sort of display box so the user can see what they're editing? Or what would be the best way to go about editing?

Reply:
Yea it sounds like you are on the right track @omdnrd.  

What I'd suggest:

1. Add someplace in the UI for a user to "edit" an existing one
2. Let them click "save" once they finish their edit and that will call the update method in the `.ts` controller. 
3. Ensure the `.ts` controller calls the proper service file's method.
4. Ensure that the `service file` method does what the delete method does by sending down the `id` and any data in the body that needs to be updated.
5. Ensure that the server patch method does what you expect it to.

I'd go ahead and give this a shot before our call at 11 and see how far you can get.